hooks/data_validation.py

```diff
@@ -76,3 +76,15 @@ def sanitize_check_result(self, errors):
         """
 
         raise NotImplementedError()
+
+    def post_fix_action(self, rule_ids):
+        """
+        Called once an individual fix has been resolved
+
+        :param rule_ids: List of rule IDs linked to the executed fix
+        """
+        self.logger.debug(f"Executing post_fix_action() for rules {rule_ids}...")
+
+    def post_fix_all_action(self):
+        """Called once all the fixes have been resolved when the "Fix All" button have been entered"""
+        self.logger.debug("Executing post_fix_all_action()...")
```

Comment:
I really like introducing a hook for the post fix action. I'm thinking now though that we can actually move the whole fix callback functionality to hook methods. This way there is the most flexibility if the fix itself needs some customization, or pre/post fix actions.

We could change these hook methods to replace the dialog functions `_resolve_all_rules` and `_resolve_rules`, these functions would just need to be passed the manager and set of rules to fix. For the `_resolve_rules` function, I have some clean up notes on that (unrelated to this change but it will make it easier to migrate it to the hook method, see below).

Also, we should pass the set of rules for fix all and use the same `post_fix_action` as when executing the fix for a subset of all the rules. This way we consolidate to one post fix action, and in case a customer has customized the set of rules and their rules do not require any update, this would save them from refreshing when they do not need to.